src/index.ts

```diff
@@ -3,9 +3,22 @@ export type GoResultSuccess<T> = { data: T; success: true };
 export type GoResultError<E extends Error = Error> = { error: E; success: false };
 export type GoResult<T, E extends Error = Error> = GoResultSuccess<T> | GoResultError<E>;
 
+export interface StaticDelayOptions {
+  type: 'static';
+  delayMs: number;
+}
+
+export interface RandomDelayOptions {
+  type: 'random';
+  minDelayMs: number;
+  maxDelayMs: number;
+}
+
 export interface GoAsyncOptions {
-  readonly retries?: number;
-  readonly timeoutMs?: number;
+  readonly retries?: number; // Number of retries to attempt if the go callback is unsuccessful.
+  readonly timeoutMs?: number; // The timeout for each attempt.
```

Comment:
Maybe this should rather be called `attemptTimeoutMs` (and the one below `totalTimeoutMs`) just to further avoid any potential confusion?

Reply:
I was thing about this, but it would cause migration to the existing consumer (airnode, api3-dao-dashboard, chainapi?...). But now that I think about it, you can just do find/replace so it's probably worth it.

Reply:
I don't think a migration would be that big of a deal

Reply:
Okey, I've done the change the naming (could have probably done it in separate PR though)